### PR TITLE
Cosserat 'update' plasticity added

### LINKED ISSUE #9067: Create Cosserat version of ComputeAdmissibleState
### Description of the enhancement or error report

ComputeAdmissibleState in TensorMechanics computes stress and inelastic quantities for standard solid mechanics: create a version that also does the Cosserat analogues.

### Rationale for the enhancement or information for reproducing the error

I need the Cosserat version.  Presently i'm going to just implement Cosserat elasticity as we don't have any plastic models with the Cosserat curvatures in them.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)

no bad impacts, only enhancement of tensor mechanics

### PR DESCRIPTION
**Pull request by WilkAndy**
MODIFICATION TO EXISTING STUFF

ComputeMultipleInelasticStress: has a new method called computeAdmissibleState that was necessary to create because of Cosserat complications, but also has allowed removal of a few lines of code

TwoParameterPlasticityStressUpdate: pq_ok user parameter added because although i previously assumed that stress=0 was admissible that's not necessarily true.  Also, a new method - setInelasticStrainIncrementAfterReturn - added because of need to override when doing Cosserat

CappedWeakPlaneCosseratStressUpdate: now derives from CappedWeakPlaneStressUpdate which allows removal of some code

ComputeLayeredCosseratElasticityTensor: now computes "compliance" which is the inverse of the elasticity tensor.  The reason for this is two-fold: (1) the usual RankTwoTensor.invSymm() does not work in this case because the elasticity tensor is unsymmetric; (2) computing compliance once in the constructor is so much cheaper than computing it on-the-fly every time it's needed.

NEW ADDITIONS

CappedDruckerPragerCosseratStressUpdate: this is the Cosserat equivalent of the CappedDruckerPrager plasticity model.  It is complicated because its flow rule uses a different elasticity tensor than the stress-strain relation.  This is the core reason behind some of the above modifications.

ComputeMultipleInelasticCosseratStress: this is the Cosserat equivalent of ComputeMultipleInelasticStress.  Currently it does only Cosserat elasticity, because the Cosserat plasticity models (CappedDruckerPragerCosseratStressUpdate and CappedWeakPlaneCosseratStressUpdate) do not involve the moment-stress in their yield functions or flow potentials.

Fixes #9067

### CODE REVIEW

**Comment by WilkAndy** [comment]
Yep PETSc debug:linux-petsc-dbg doesn't like that preconditioners/pbp.lots_of_variables test.

File: modules/tensor_mechanics/src/materials/CappedDruckerPragerCosseratStressUpdate.C [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    RankFourTensor inv = RankFourTensor(RankFourTensor::initIdentitySymmetricFour) + Tijab;
+    try
+    {
+      inv = inv.transposeMajor().invSymm();
+    }
+    catch (const MooseException & e)
+    {
+      // Cannot form the inverse, so probably at some degenerate place in stress space.
+      // Just return with the "best estimate" of the cto.
+      return;
```
Comment (permcody) [inline modules/tensor_mechanics/src/materials/CappedDruckerPragerCosseratStressUpdate.C]: Not even a warning? I guess if this is common that's fine but are you not interested at all if it fails?

**Review by permcody — DISMISSED** [review]
Looks fine to me.

File: modules/tensor_mechanics/src/materials/CappedDruckerPragerCosseratStressUpdate.C (hunk 1 above)
Comment (WilkAndy) [inline modules/tensor_mechanics/src/materials/CappedDruckerPragerCosseratStressUpdate.C]: Warning - good idea.  This is a very rare occurrence (i've never been able to devise a test for it actually!)  I've included the mooseWarning

**Comment by WilkAndy** [comment]
Many unrelated failures in RattleSnake, all seem to be caused by "A 'SteadyState' is not a registered object"  I'm not sure who the RattleSnake developers are so i can't notify them.

**Comment by permcody** [comment]
That's be Yaqi

**Comment by WilkAndy** [comment]
@YaqiWang - lots of failures in RattleSnake testing due to "A 'SteadyState' is not a registered object"

a

**Comment by WilkAndy** [comment]
This closes #7243

**Comment by permcody** [comment]
Failure is timing and unrelated to this PR. Merging.

**Comment by WilkAndy** [comment]
Thanks for the merge, @permcody,

a

**Pull request merged**